app_test.go

```diff
@@ -2183,3 +2183,92 @@ func TestSetupInitializesOnlyNilWriters(t *testing.T) {
 		t.Errorf("expected a.Writer to be os.Stdout")
 	}
 }
+
+func TestApp_PreventDefaultPrint_When_BoolFlag(t *testing.T) {
+	var actualOutput bytes.Buffer
+
+	expectedOutputSlice := strings.Fields(`NAME:
+	greet - fight the loneliness!
+	   
+  USAGE:
+	cli.test [global options] command [command options] [arguments...]
+  
+  COMMANDS:
+	help, h  Shows a list of commands or help for one command
+  
+  GLOBAL OPTIONS:
+	--verbose, -v  Increase verbosity (default: default text)
+	--extract, -e  Extract only, do not execute the service
+	--keep, -k     Do not delete the temp dir after execution
+	--help, -h     show help`)
+
+	app := App{
+		Name:   "greet",
+		Usage:  "fight the loneliness!",
+		Writer: &actualOutput,
+		Flags: []Flag{
+			&BoolFlag{
+				Name:             "verbose",
+				Aliases:          []string{"v"},
+				Usage:            "Increase verbosity",
+				ShowDefaultValue: true,
+				DefaultText:      "default text",
+			},
+			&BoolFlag{
+				Name:             "extract",
+				Aliases:          []string{"e"},
+				Usage:            "Extract only, do not execute the service",
+				ShowDefaultValue: false,
+			},
+			&BoolFlag{
+				Name:    "keep",
+				Aliases: []string{"k"},
+				Usage:   " Do not delete the temp dir after execution",
+			},
+		},
+
+		Action: func(c *Context) error {
+			return nil
+		},
+	}
+
+	_ = app.Run([]string{"", "-h"})
```

Comment:
Can you also check to ensure this error is not nil?

Reply:
pls check now